src/rdb.c

```diff
@@ -2185,6 +2217,161 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
 
         /* All pairs should be read by now */
         serverAssert(len == 0);
+    } else if (rdbtype == RDB_TYPE_HASH_METADATA) {
+
+        size_t fieldLen;
+        sds value, field;
+        uint64_t expire, minExpire = EB_EXPIRE_TIME_INVALID;
+        mstime_t now = mstime();
+        dict *dupSearchDict = NULL;
+        ebuckets *hexpires = (db != NULL ? &db->hexpires : NULL);
+
+        len = rdbLoadLen(rdb, NULL);
+        if (len == RDB_LENERR) return NULL;
+        if (len == 0) goto emptykey;
+        o = createHashObject();
+        /* Too many entries? Use a hash table right from the start. */
+        if (len > server.hash_max_listpack_entries) {
+            hashTypeConvert(o, OBJ_ENCODING_HT, hexpires);
+            dictTypeAddMeta((dict**)&o->ptr, &mstrHashDictTypeWithHFE);
+            initDictExpireMetadata(key, o);
+        } else {
+            hashTypeConvert(o, OBJ_ENCODING_LISTPACK_EX, hexpires);
+            if (deep_integrity_validation) {
+                /* In this mode, we need to guarantee that the server won't crash
+                * later when the listpack is converted to a dict.
+                * Create a set (dict with no values) for dup search.
+                * We can dismiss it as soon as we convert the listpack to a hash. */
+                dupSearchDict = dictCreate(&hashDictType);
+            }
+        }
```

Comment:
maybe we can create hashtable or listpackex directly, instead of creating a listpack then converting to hashtable or listpackex.
may a TODO comment for future optimize.